library/components/TTable.vue

```diff
@@ -0,0 +1,1018 @@
+        );
+      }
+
+      // can_collapse_detail_rows
+      if (this.onlyShowOneDetailRow && !this.can_collapse_detail_rows) {
+        console.warn(
+          "The 'onlyShowOneDetailRow' prop is only valid when the 'can_collapse_detail_rows' is true"
+        );
+      }
+
+      // selected items
+      if (this.selectedItems) {
+        const selectedItemsNotInRows = _.difference(
+          this.selectedItems,
+          this.uniqueOriginalRows
+        );
+        if (selectedItemsNotInRows.length > 0) {
+          console.warn(
+            "These selected items are not in the rows: ",
+            selectedItemsNotInRows
+          );
+        }
+      }
+
+      // searching
+      if (
+        this.can_search &&
+        !this.enableSearchFilter &&
+        !this.enableSearchHighlight
+      ) {
+        console.warn(
+          "The search input is visible but both the enableSearchFilter and the enableSearchHighlight props are disabled so searching will have no effect"
+        );
+      }
+
+      // sorting
+      let sortableColumns = this.internalColumns.filter((c) => {
+        return c.sort;
+      });
+      if (sortableColumns.length > 0) {
+        sortableColumns = _.sortBy(sortableColumns, ["sort.priority"]);
+        let lastSortableColumn = null;
+        sortableColumns.forEach((column) => {
+          const sortOptions = column.sort;
+
+          if (typeof sortOptions.priority !== "number") {
+            console.warn(
+              "The sort priority option must be a number. For column " +
+                column.property +
+                " it is a " +
+                typeof sortOptions.priority
+            );
+          }
+          if (!["asc", "desc"].includes(sortOptions.direction)) {
+            console.warn(
+              "The sort direction option must be either 'asc' or 'desc'. For column " +
+                column.property +
+                " it is " +
+                sortOptions.direction
+            );
+          }
+          if (
+            lastSortableColumn !== null &&
+            lastSortableColumn.sort.priority === column.sort.priority
+          ) {
+            console.warn(
+              "The sort priority option for column " +
+                column.property +
+                " is the same as the sort priority option for " +
+                lastSortableColumn.property
+            );
+          }
+          lastSortableColumn = column;
+        });
+      }
+    },
+    expandRow(row) {
+      if (this.onlyShowOneDetailRow) {
+        this.internalRows.forEach((row) => (row.detailRowOpen = false));
+      }
+      row.detailRowOpen = true;
+      const index = this.internalRows.indexOf(row);
+      this.$set(this.internalRows, index, row);
+    },
+    collapseRow(row) {
+      row.detailRowOpen = false;
+      const index = this.internalRows.indexOf(row);
+      this.$set(this.internalRows, index, row);
```

Comment:
OH that's why the layer array was an object in EM! I was so confused! Going to leave it for now since this needs to continue to work with Vue2 until the core is updated, but adding a comment to not bother when that upgrade is made.